src/Sparrow/Json/JsonOperationContext.cs

```diff
@@ -1004,7 +1016,7 @@ protected internal virtual unsafe void Reset(bool forceReleaseLongLivedAllocator
                 {
                     _arenaAllocatorForLongLivedValues.Return(mem.AllocatedMemoryData);
                     mem.AllocatedMemoryData = null;
-                    mem.Dispose();
+                    mem.Dispose(); // todo: probably we don't need this call
```

Comment:
why not?

Reply:
in the dispose function we do:
1. throw exception if already disposed (we don't really care about it here)
2. return memory to the context (we just did it)
3. raise the IsDisposed flag for the LSV (actually, this one is the more important part..)

What we really want to do is to raise the IsDisposed flag from outside (I'll do just that)